File: AgentApp/Frontend/src/app/model/advertisement.ts
import { PriceList } from 'src/app/model/priceList';
import { Car } from 'src/app/model/car';
import { DateTime } from 'luxon';
export class Advertisement {
    id: number;
    car: Car;
    priceList: PriceList;
    kilometresLimit: number;
    discount: number;
    cdw: boolean;
    dateFrom: string;
    constructor(car: Car, priceList: PriceList, discount: number, kilometresLimit: number, cdw: boolean, dateFrom: string, id?: number) {
        this.id = id;
        this.car = car;
        this.priceList = priceList;
        this.discount = discount;
        this.kilometresLimit = kilometresLimit;
        this.cdw = cdw;
        this.dateFrom = dateFrom;
    }

}

File: MainApp/SearchService/src/main/java/jvn/SearchService/model/Advertisement.java
package jvn.SearchService.model;

import jvn.SearchService.enumeration.LogicalStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Advertisement {

    @Id
    private Long id;

    @Column(nullable = false)
    private LogicalStatus logicalStatus = LogicalStatus.EXISTING;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Car car;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private PriceList priceList;

    @Column(nullable = false)
    private LocalDate dateFrom;

    @Column
    private LocalDate dateTo;

    @Column
    private Integer kilometresLimit;

    @Column
    private Integer discount;

    @Column
    private Boolean CDW;

    @Column
    private String pickUpPoint;
}
